bin/i18n/sync-up.rb

```diff
@@ -9,6 +9,8 @@
 require_relative 'metrics'
 require_relative 'i18n_script_utils'
 
+Dir[File.expand_path('../resources/*.rb', __FILE__)].sort.each {|file| require file}
```

Comment:
I feel like I've asked this before but why is this `require` at the beginning of the script instead of the end? Is it because we can execute the file?

Reply:
We don't have to require them at the end of the class because these files do not depend on any data from the `sync-up`. 

For example:

We need to require `i18n/resources/dashboard/base_content.rb` after we define the `Dashboard` module in `i18n/resources/dashboard.rb`
https://github.com/code-dot-org/code-dot-org/blob/815b83c85efde69df6dde017fb21f355a63388fb/bin/i18n/resources/dashboard.rb#L53-L55
to define (load into the memory) the `Dashboard` module data (`ORIGIN_I18N_DIR_PATH` and `I18N_SOURCE_DIR_PATH`) first
https://github.com/code-dot-org/code-dot-org/blob/c7dbf69132b9de2ec7ac7e06894fd89dac2da5ee/bin/i18n/resources/dashboard.rb#L7-L8
so that then we can use it to define the `BaseContent` class data (`ORIGIN_I18N_FILE_PATH` and `I18N_SOURCE_FILE_PATH`)
https://github.com/code-dot-org/code-dot-org/blob/c7dbf69132b9de2ec7ac7e06894fd89dac2da5ee/bin/i18n/resources/dashboard/base_content.rb#L12-L13